src/NodeApi/Interop/JSModuleContext.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.JavaScript.NodeApi.Interop;
+
+/// <summary>
+/// Manages JavaScript interop context for the lifetime of a .NET module.
+/// </summary>
+/// <remarks>
+/// A <see cref="JSModuleContext"/> instance is constructed when the module is loaded and disposed
+/// when the module is unloaded.
+/// </remarks>
+public sealed class JSModuleContext : IDisposable
+{
+    /// <summary>
+    /// Gets the current module context.
+    /// </summary>
+    public static JSModuleContext Current
+    {
+        get
+        {
+            JSValueScope scope = JSValueScope.Current
+                ?? throw new InvalidCastException("No current scope.");
+            return scope.ModuleContext
+                ?? throw new InvalidCastException("No current module context.");
+        }
+    }
+
+    /// <summary>
+    /// Gets an instance of the class that represents the module, or null if there is no module
+    /// class.
+    /// </summary>
+    public object? Module { get; internal set; }
+
+    public bool IsDisposed { get; private set; }
+
+    public void Dispose()
+    {
+        if (IsDisposed) return;
+
+        IsDisposed = true;
+
+        if (Module is IDisposable module)
+        {
+            module.Dispose();
+        }
+
+        GC.SuppressFinalize(this);
```

Comment:
>GC.SuppressFinalize(this); [](http://example.com/codeflow?start=8&length=26)

I wonder if it is needed when we have no class Finalizer.

Reply:
The doc says it has no effect on an object that does not have a finalizer.

We have it in ~10 places in the codebase, but I think none of those are needed except for `JSReference` which does define a finalizer.

Reply:
Cleaned up.